OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -1316,25 +1218,35 @@ public static void setEmail(@NonNull final String email, @Nullable final String
     *                      Omit this value if you do not have a backend to authenticate the user.
     * @param callback Fire onSuccess or onFailure depending if the update successes or fails
     */
-   public static void setEmail(@NonNull final String email, @Nullable final String emailAuthHash, @Nullable EmailUpdateHandler callback) {
-
-      //if applicable, check if the user provided privacy consent
+   public static void setEmail(@NonNull final String email, @Nullable final String emailAuthHash, @Nullable final EmailUpdateHandler callback) {
+      if (taskController.shouldQueueTaskForInit("setEmail()")) {
```

Comment:
These OneSignal public methods now seem to go through these steps.
1. Check if remote params are done
2. If not add to queue
3. After remote params finishes check if privacy consent is required.
4. Check if OneSignal has been initialized


However with step 4 if we got here we already download remote params so OneSignal has been initialized. So this is a redundant check.

Reply:
yeah agree 4th step could be removed

Reply:
Some odd reason removing step 4 causing the test on the method to stall. Noticed this `setExternalUserId` it gets stuck on `threadAndTaskWait` but probably for other methods too

Reply:
Solved this, the reason was It was only queuing runnable not pending runnable, and because of that, Executor was not being finished